executor/prepared.go

```diff
@@ -325,3 +326,39 @@ func CompileExecutePreparedStmt(ctx context.Context, ID uint32, args ...interfac
 	}
 	return sa
 }
+
+// ResetStmtCtx resets the StmtContext.
+// Before every execution, we must clear statement context.
+func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
+	sessVars := ctx.GetSessionVars()
+	sc := new(variable.StatementContext)
+	switch s.(type) {
+	case *ast.UpdateStmt, *ast.InsertStmt, *ast.DeleteStmt:
+		sc.IgnoreTruncate = false
```

Comment:
This PR just move the old code to another package.
We better not moving code and modify it at the same time.